pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/signature/JavaFieldSignature.java

```diff
@@ -54,10 +55,10 @@ private static int code(Visibility visibility, boolean isStatic, boolean isFinal
      *
      * @return The signature of the field
      */
-    public static FieldSignature buildFor(ASTFieldDeclaration node) {
+    public static JavaFieldSignature buildFor(ASTFieldDeclaration node) {
         int code = code(Visibility.get(node), node.isStatic(), node.isFinal());
```

Comment:
Hmm... We need to make sure that every possible combination has a different code. Maybe you can write a unit test, that collects all possible instances in a set and then verify the size. If we hit a collision, we will be missing some instances.